### PR TITLE
jwt_get_grants & jwt_get_headers

### PR DESCRIPTION
**Pull request by RekGRpth**
to get nested json in https://github.com/RekGRpth/nginx-jwt-module/commit/c9c2e9b1a9846d7c7d0c3542248f9c418dadf990

### CODE REVIEW

**Comment by benmcollins** [comment]
Thanks for the code, however, I'm going to reject this for several reasons:

1) There's no code to test and cover these new functions
2) There's no documentation for the new functions
3) I do not want to export JANSSON to LibJWT users. If I ever wanted to make use of alternative JSON libraries or do away with it altogether, this would not be possible after accepting this commit. Note how I do no export OpenSSL/GnuTLS, so that it is abstracted to the library interfaces.
4) Users of LibJWT can easily implement this functionality themselves, if they need it.

Truth is, I've been looking at ways of limiting or even removing the use of JANSSON for a lot of reasons.

**Comment by RekGRpth** [comment]
How I can get value by sequence of nested json keys?

**Comment by RekGRpth** [comment]
May be something like
```c
JWT_EXPORT char *jwt_get_grants_json(jwt_t *jwt, ...);
```
(va_args)?

**Comment by benmcollins** [comment]
Might be easier to answer if you show me an example of what you are trying to do.

**Comment by RekGRpth** [comment]
for example
```c
jwt_get_grants_json(jwt, NULL)
```
returns
```json
{"a":{"b":"c"}}
```
and I need to get "c" by keys "a" and "b"

**Comment by benmcollins** [comment]
If you have the JSON string, then you can use whatever parser you want (e.g. JANSSON) to get to the nested value.

My main point is that LibJWT is meant to generate and validate JWT strings. It's not meant to be a JSON parser. I've built what I can into it so that the user can get what they need and build what they want. If I go down the road of providing an abstracted JSON interface, then there's nothing stopping the additional request to add even more JSON interfaces, and that's a road I don't want to go down.

Basically, I want to let LibJWT do what is was designed for, and let other libraries handle what they are good at, and not replace them.

**Comment by RekGRpth** [comment]
How about something like
```c
const char *jwt_get_grant_n(jwt_t *jwt, int count, ...) {
    errno = EINVAL;
    if (!jwt || !count) return NULL;

    json_t *js_val = jwt->grants;
    va_list args;
    va_start(args, count);

    for (int i = 0; i < count; i++) {
        const char *grant = va_arg(args, const char *);
        if (!grant || !strlen(grant)) return NULL;
        js_val = json_object_get(js_val, grant);
        if (js_val == NULL) return NULL;
    }

    va_end(args);
    errno = 0;

    return json_string_value(js_val);
}
```
to get nested json from
```json
{"a":{"b":"c"}}
```
```c
const char *s = jwt_get_grant_n(jwt, 2, "a", "b");
```
